src/Credentials/CredentialProvider.php

```diff
@@ -61,11 +62,16 @@ class CredentialProvider
      */
     public static function defaultProvider(array $config = [])
     {
+        $cache = isset($config['credentials'])
```

Comment:
Instead of two ternary checks here, could we just create an `$instanceProfileProvider` variable that is first created as a stock instance profile provider and then decorated with a cache if the cache parameter is present? I think that would make this easier to follow.